src/reporters/udp_sender.js

```diff
@@ -146,23 +151,18 @@ export default class UDPSender {
     this._reset();
 
     if (writeResult.err) {
-      this._logger.error(`error writing Thrift object: ${writeResult.err}`);
-      if (callback) {
-        callback({ err: true, numSpans: numSpans });
-      }
+      this._invokeCallback(callback, numSpans, `error writing Thrift object: ${writeResult.err}`);
       return;
     }
 
     // Having the error callback here does not prevent uncaught exception from being thrown,
     // that's why in the constructor we also add a general on('error') handler.
     this._client.send(thriftBuffer, 0, thriftBuffer.length, this._port, this._host, (err, sent) => {
       if (err) {
-        this._logger.error(
-          `error sending spans over UDP: ${err}, packet size: ${writeResult.offset}, bytes sent: ${sent}`
-        );
-      }
-      if (callback) {
-        callback({ err: err !== 0, numSpans: numSpans });
+        let error: string = err && `error sending spans over UDP: ${err}, packet size: ${writeResult.offset}, bytes sent: ${sent}`;
```

Comment:
I'm surprised this is not invalid. The type should actually be `?string`, I think.

Looks like the reason it's not invalid _might_ be the version of flow being used is pretty ancient, 0.30.0. [Link to the error](https://flow.org/try/#0PQKgBAAgZgNg9gdzCYAoVBjOA7AzgFzBgFMA1OASwBMAuMAfgENsBPMAXjAFdsrioK2YlQDcmHATAVcAeQDWdegQBOggOYciZSlTAAyPWADkcALZqjYrHkLSAgtgCiy5XGV0V6zSXLV9hk3NLIA).

And, I recommend `const`, here.